sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala

```diff
@@ -441,6 +441,9 @@ private[hive] class HiveMetastoreCatalog(sparkSession: SparkSession) extends Log
       case p: LogicalPlan if p.resolved => p
 
       case p @ CreateHiveTableAsSelectLogicalPlan(table, child, allowExisting) =>
+        // Ensuring whether no duplicate name is used in table definition
+        checkDuplicates(child.output.map(_.name), s"table definition of ${table.identifier}")
```

Comment:
`PreWriteCheck` is checked after conversion from `CreateHiveTableAsSelectLogicalPlan ` to `execution.CreateHiveTableAsSelectCommand`. `PreWriteCheck` is unable to access the Hive package `execution.CreateHiveTableAsSelectCommand`. Thus, I have no clue how to move this into `PreWriteCheck `. Introduce a new rule?

Actually, in Hive, there is a stage called `Semantic Analysis`, which is done before `Analyzer` but after `Parser`. That stage is for checking these semantic errors. Not sure we should add a similar concept into Spark SQL.